Esri/arcgis-maps-sdk-swift-samples
# [Update] Use model to manage geo view and data

## Description

This PR changes the `@StateObject` pattern in most of the samples, to use an observable model object to manage the data.

## Linked Issue(s)

- `swift/issues/3271`
- `swift/issues/3250`
- https://github.com/ArcGIS/arcgis-maps-sdk-swift-toolkit/issues/200
- Potentially #48 and #89 

## How To Test

To run the app, you'll need to

1. replace the toolkit dependency line in Package.swift locally with `.package(name: "arcgis-maps-sdk-swift", path: "../swift/ArcGIS")`.
2. drag the toolkit into the samples project, so it overrides the remote resolved packages and uses the local toolkit repo instead.

No behavioral change should present. Verify all changed sample to work as expected.

## To Discuss

- What should be a good generic docstring for the view model objects?
